flake.nix

```diff
@@ -0,0 +1,68 @@
+{
+  description = "ogmios-datum-cache";
+
+  inputs = {
+    haskell-nix = {
+      url = "github:input-output-hk/haskell.nix";
```

Comment:
I'd recommend avoiding upstream's haskell.nix, with two possible solutions:

- just use nixpkgs' Haskell infrastructure. I'd suggest this since everything is cached reliably
- use the MLabs fork of haskell.nix. We don't actually need to do this, since we don't rely on anything extra dependencies in the `cabal.project`